integration-tests/src/tests/network/runner.rs

```diff
@@ -127,24 +124,120 @@ pub enum Action {
     // Send stop signal to some node.
     Stop(usize),
     // Wait time in milliseconds
-    Wait(usize),
-    #[cfg(feature = "test_features")]
+    Wait(time::Duration),
+    #[allow(dead_code)]
     SetOptions {
         target: usize,
         max_num_peers: Option<u64>,
     },
 }
 
-#[derive(Clone)]
 pub struct RunningInfo {
-    pm_addr: Vec<Addr<PeerManagerActor>>,
+    runner: Runner,
+    nodes: Vec<Option<NodeHandle>>,
     peers_info: Vec<PeerInfo>,
 }
 
 struct StateMachine {
     actions: Vec<ActionFn>,
 }
 
+async fn check_routing_table(
```

Comment:
check_* async functions are polling the state of the system-under-test and waiting until a predicate is fulfilled. anyhow::Result is used for unexpected errors ~originating from the code under test. But I rather use Result here specifically so that I don't have to read the over-verbose tokio stack trace.